Improve Heap Histogram docs (#34237)
DataDog/documentation: content/en/profiler/profiler_troubleshooting/java.md

[Learn how to use override templates.](#creating-and-using-a-jfr-template-override-file)

## Enabling the heap histogram collection
<div class="alert alert-info">This feature requires at least Java 17.0.9 or newer and does not work with ZGC</div>

> To enable the heap histogram collection, start your application with the `-Ddd.profiling.heap.histogram.enabled=true` JVM setting or the `DD_PROFILING_HEAP_HISTOGRAM_ENABLED=true` environment variable. This powers the Heap Occupancy metrics as well as improves the Memory Leaks workflow.

Reviewer: Should we link to https://docs.datadoghq.com/profiler/guide/solve-memory-leaks/ here?

> To enable the heap class histogram collection, start your application with the `-Ddd.profiling.heap.histogram.enabled=true` JVM setting or the `DD_PROFILING_HEAP_HISTOGRAM_ENABLED=true` environment variable. This powers the Heap Occupancy metrics as well as improves the Memory Leaks workflow.

Reviewer 2: ```suggestion
To enable the heap class histogram collection, start your application with the `-Ddd.profiling.heap.histogram.enabled=true` JVM setting or the `DD_PROFILING_HEAP_HISTOGRAM_ENABLED=true` environment variable. This powers the Heap Occupancy metrics and improves the Memory Leaks workflow.
```
